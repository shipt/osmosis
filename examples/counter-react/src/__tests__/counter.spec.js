import React from 'react';
import Counter from '../counter';
import { CounterContext } from '../store';
import { render, fireEvent } from '@testing-library/react';

describe('Counter', () => {
  const CounterStore = {
    state: {
      count: 0
    },
    increment: jest.fn(),
    decrement: jest.fn()
  };
  let ContextComponent = () => (
    <CounterContext.Provider value={[CounterStore]}>
      <Counter />
    </CounterContext.Provider>
  );
  it('renders default', async () => {
    let wrapper = await render(<ContextComponent />);
    expect(wrapper.getByTestId('counter-wrap')).toMatchSnapshot();
  });
  it('tests increment button', async () => {
    let wrapper = await render(<ContextComponent />);
    fireEvent.click(wrapper.getByTestId('increment'));
    expect(CounterStore.increment).toHaveBeenCalled();
  });
  it('tests decrement button', async () => {
    let wrapper = await render(<ContextComponent />);
    fireEvent.click(wrapper.getByTestId('decrement'));
    expect(CounterStore.decrement).toHaveBeenCalled();
  });
});
