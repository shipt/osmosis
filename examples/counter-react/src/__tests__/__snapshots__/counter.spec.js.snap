// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders default 1`] = `
<div
  data-testid="counter-wrap"
>
  <p>
    Count: 
    0
  </p>
  <button
    data-testid="decrement"
  >
    -
  </button>
  <button
    data-testid="increment"
  >
    +
  </button>
</div>
`;
